### PR TITLE
chore(fallback): projcet version check

### PR DESCRIPTION
**Pull request by xzf0587**

### CODE REVIEW

**Review by Alive-Fish — DISMISSED** [review]

**Review by jayzhang — DISMISSED** [review]

**Review by Alive-Fish — DISMISSED** [review]

**Review by jayzhang — APPROVED** [review]

File: packages/fx-core/src/common/versionMetadata.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+export const MetadataV3 = {
+  projectVersion: "1.0.0",
+  platformVersion: {
+    vs: "17.5.x.x",
```
Comment (adashen) [inline packages/fx-core/src/common/versionMetadata.ts:7]: is this pattern only work for 17.5 preview?

**Review by Alive-Fish — APPROVED** [review]

File: packages/fx-core/src/common/versionMetadata.ts (hunk 1 above)
Comment (xzf0587) [reply packages/fx-core/src/common/versionMetadata.ts]: The value is not used in vs 17.5 fallback api.
Will update the value in dev branch and remove it here.

File: packages/fx-core/src/common/versionMetadata.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  localConfigFile: "teamsapp.local.yml",
+  defaultEnvironmentFolder: "teamsAppEnv",
+  projectId: "projectId",
+};
+
+export const MetadataV2 = {
+  projectVersion: "2.0.0",
+  projectMaxVersion: "2.1.0",
+  platformVersion: {
+    vs: "17.4.x.x",
```
[the remark this answers is not in the document]
Comment (xzf0587) [reply packages/fx-core/src/common/versionMetadata.ts]: The value is not used in vs 17.5 fallback api.
Will update the value in dev branch and remove it here.

File: packages/fx-core/src/core/FxCore.ts [hunk 3]
```diff
@@ -1471,6 +1474,20 @@ export class FxCore implements v3.ICore {
     return ok(Void);
   }
 
+  // a project version check
+  async projectVersionCheck(inputs: Inputs): Promise<Result<VersionCheckRes, FxError>> {
```
Comment (blackchoey) [inline packages/fx-core/src/core/FxCore.ts:1478]: What's the scope of this function? Is it just tell whether it's a V2 project (versionInfo.isSupported = compatible) or not (versionInfo.isSupported = unsupported)?

Comment (xzf0587) [reply packages/fx-core/src/core/FxCore.ts:1478]: In vs 17.5, no project is upgradable, versionInfo.isSupported could be compatible for V2 project and unsupported for V3 project.

**Review by blackchoey — APPROVED** [review]

**Review by adashen — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

**Pull request merged**